bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/spinner/SpinnerAjaxButton.java

```diff
@@ -90,21 +91,32 @@ public LaddaAjaxButton setEffect(LaddaBehavior.Effect effect) {
      * @param color The color for the spinner
      * @return {@code this}, for chaining
      */
-    public LaddaAjaxButton setSpinnerColor(LaddaBehavior.Color color) {
-        this.laddaBehavior.withSpinnerColor(color);
+    public SpinnerAjaxButton setSpinnerColor(SpinnerBehavior.Color color) {
+        this.spinnerBehavior.withSpinnerColor(color);
+        return this;
+    }
+
+    /**
+     * Sets if label should be hidden
+     *
+     * @param noLabel <code>true</code> if label should be hidden
+     * @return {@code this}, for chaining
+     */
+    public SpinnerAjaxButton setEffect(boolean noLabel) {
+        this.noLabel = noLabel;
         return this;
     }
 
     @Override
     protected void updateAjaxAttributes(AjaxRequestAttributes attributes) {
         super.updateAjaxAttributes(attributes);
-        attributes.getAjaxCallListeners().add(new LaddaAjaxCallListener());
+        attributes.getAjaxCallListeners().add(new SpinnerAjaxCallListener());
     }
 
     @Override
     protected <L extends Serializable> Component newLabel(String markupId, IModel<L> model) {
         Component label = super.newLabel(markupId, model);
-        if (model.getObject() == null || "".equals(model.getObject())) {
+        if (noLabel) {
```

Comment:
I like the previous version better, i.e. decide whether to show/hide the label depending on the model object.
You can use `Strings.isEmpty(component.getDefaultModelObjectAsString())`. See de.agilecoders.wicket.core.util.Components#hideIfModelIsEmpty

Reply:
The old behavior restored
Thanks for the pointer